appinventor/components/src/com/google/appinventor/components/runtime/Ev3Mailbox.java

```diff
@@ -0,0 +1,69 @@
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.SimpleFunction;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.runtime.util.Ev3BinaryParser;
+import com.google.appinventor.components.runtime.util.Ev3Constants;
+import com.google.appinventor.components.common.YaVersion;
+import java.util.*;
+
+/**
+ * A component that provides a high-level interface to mailbox communication with a
+ * LEGO MINDSTORMS EV3 robot.
+ *
+ * @author tasosggps@gmail.com (Gkagkas Anastasios)
+ */
+
+@DesignerComponent(version = YaVersion.EV3_MOTORS_COMPONENT_VERSION,
+        description = "A component that provides high level interfaces to a LEGO MINDSTORMS EV3 " +
+                "robot, with functions that can send messages to the Ev3 mailbox.",
+        category = ComponentCategory.LEGOMINDSTORMS,
+        nonVisible = true,
+        iconName = "images/legoMindstormsEv3.png")
+@SimpleObject
+public class Ev3Mailbox extends LegoMindstormsEv3Base{
```

Comment:
Code style: there should be a space before the opening `{`. Apply this to the other occurrences in this file.